spec/unit/helpers/check_helper_spec.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+require 'gooddata/lcm/actions/base_action'
+require 'gooddata/lcm/helpers/check_helper'
+require 'gooddata/lcm/types/types'
+
+describe 'GoodData::LCM2::Helpers::Check' do
+  let(:params) do
+    params = {
+      test_param_two: 'Testing param two',
+      test_param_three: 'Testing param three',
+      test_param_four: 4
+    }
+    GoodData::LCM2.convert_to_smart_hash(params)
+  end
+  it 'verifies required' do
+    PARAMS = GoodData::LCM2::BaseAction.define_params(self) do
+      description 'Testing param one'
+      param :test_param_one, instance_of(GoodData::LCM2::Type::StringType), required: true
+    end
+    expect { GoodData::LCM2::Helpers.check_params(PARAMS, params) }.to raise_error(/Mandatory/)
+  end
+  it 'checks types' do
+    PARAMS_3 = GoodData::LCM2::BaseAction.define_params(self) do
```

Comment:
Similar code found in check_helper_spec.rb:27, check_helper_spec.rb:35 (mass = 40)